### PR TITLE
[5.0] Replace hard coded img element by JLayout joomla.html.image and DRY

### PR DESCRIPTION
**Pull request by hans2103**
Pull Request for no Issue

### Summary of Changes

This PR will change the hard coded `
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: modules/mod_banners/tmpl/default.php [hunk 1]
```diff
@@ -39,6 +40,17 @@
                 <?php $alt = $item->params->get('alt'); ?>
                 <?php $alt = $alt ?: $item->name; ?>
                 <?php $alt = $alt ?: Text::_('MOD_BANNERS_BANNER'); ?>
+                <?php $imageAttributes = [
+                    'src' => $baseurl . $imageurl,
+                    'alt' => htmlspecialchars($alt, ENT_QUOTES, 'UTF-8')
```
Comment (dgrammatiko) [inline modules/mod_banners/tmpl/default.php]: The `alt` is escaped in the Layout so you could pass it directly here

Comment (hans2103) [reply modules/mod_banners/tmpl/default.php]: adjusted my PR

File: modules/mod_banners/tmpl/default.php [hunk 2]
```diff
@@ -39,6 +40,17 @@
                 <?php $alt = $item->params->get('alt'); ?>
                 <?php $alt = $alt ?: $item->name; ?>
                 <?php $alt = $alt ?: Text::_('MOD_BANNERS_BANNER'); ?>
```
Comment (brianteeman) [inline modules/mod_banners/tmpl/default.php:40]: Does this line ever get reached. Surely the previous condition is always matched

Comment (hans2103) [reply modules/mod_banners/tmpl/default.php:40]: I was wondering the same thing. I think that is out of scope of this PR and adjusting / removing the line should be in another PR. Agree?

Comment (brianteeman) [reply modules/mod_banners/tmpl/default.php:40]: yes I realised that after typing it ;)

Comment (hans2103) [reply modules/mod_banners/tmpl/default.php:40]: should be something like below I guess

```
$alt = $item->params->get('alt', $item->name); 
```

Comment (hans2103) [reply modules/mod_banners/tmpl/default.php:40]: but... that's up to another PR. ;-)

File: modules/mod_banners/tmpl/default.php [hunk 3]
```diff
@@ -39,6 +40,17 @@
...
+                <?php $imageAttributes = [
```
Comment (wojtekLs) [inline modules/mod_banners/tmpl/default.php:43]: why there are multiple `php` block there?

File: modules/mod_banners/tmpl/default.php [hunk 4]
```diff
@@ -47,16 +59,7 @@
...
-                                src="<?php echo $baseurl . $imageurl; ?>"
-                                alt="<?php echo htmlspecialchars($alt, ENT_QUOTES, 'UTF-8'); ?>"
-                                <?php if (!empty($width)) {
-                                    echo 'width="' . $width . '"';
-                                } ?>
-                                <?php if (!empty($height)) {
-                                    echo 'height="' . $height . '"';
-                                } ?>
-                            >
+                            <?php echo $image; ?>
```
Comment (wojtekLs) [inline modules/mod_banners/tmpl/default.php:62]: LGTM :)

**Review by wojtekLs — CHANGES_REQUESTED** [review]
Please merge multiple `php` blocks into one. Same goes with `if` statements.

File: modules/mod_banners/tmpl/default.php (hunk 3 above)
Comment (richard67) [reply modules/mod_banners/tmpl/default.php:41]: Because we do it like that in our layouts so we have an HTML code style with small PHP blocks.

**Comment by richard67** [comment]
> Please merge multiple `php` blocks into one. Same goes with `if` statements.

Maybe you should first study our code style before making recommendations as if you were a long term contributor.

**Review by dgrammatiko — APPROVED** [review]

**Review by chmst — APPROVED** [review]

**Comment by chmst** [comment]
Looks good to me. The $alt seems to be not correct. but this is not in scope of this PR.

**Comment by richard67** [comment]
> > code style
> 
> @richard67 No man. Having formalized code style doesnt mean that its good code style. Nor does it mean that it should be blindly followed.

@wojtekLs Better than following your comments which are completely useless since you talk about stuff of which you have no knowledge, e.g. our rules.

**Comment by richard67** [comment]
https://developer.joomla.org/coding-standards/php-code.html

> Mixed language usage (e.g. at the layout files)
>
> For layout files and all files where we use a mix of PHP and HTML (all PHP files in the view/tmpl and layout folder) we additionally wrap every line into a <?php ... ?> block and use the alternative syntax for control structures. This should make the code easier to read and make it easier to move blocks around without creating fatal errors due to missing <?php ... ?> tags.

@wojtekLs I haven't made these rules. For changing these rules it needs a decision by the production department. We all know that, only you insist on violating these rules because you comment here without having made yourself familiar with our rules.

**Comment by brianteeman** [comment]
Why do we have to put to with this troll

**Comment by hans2103** [comment]
Please, can we get the focus on the PR again? The discussion about code styling is out of scope for this issue.
Thank you

File: modules/mod_banners/tmpl/default.php [hunk 5]
```diff
@@ -31,14 +32,23 @@
...
+                <?php $imageAttributes = [
+                    'src' => $baseurl . $imageurl,
+                    'alt' => $alt
+                ]; ?>
+                <?php if (!empty($item->params->get('width'))) {
+                    $imageAttributes['width'] = $item->params->get('width');
+                } ?>
+                <?php if (!empty($item->params->get('height'))) {
+                    $imageAttributes['height'] = $item->params->get('height');
+                } ?>
```
Comment (Quy) [inline modules/mod_banners/tmpl/default.php]:
```suggestion
                <?php if (!empty($item->params->get('width'))) : ?>
                    <?php $imageAttributes['width'] = $item->params->get('width'); ?>
                <?php endif; ?>
                <?php if (!empty($item->params->get('height'))) : ?>
                    <?php $imageAttributes['height'] = $item->params->get('height'); ?>
                <?php endif; ?>
```

**Comment by Quy** [comment]
The width/height settings are ignored.

Before PR:
```
<img
                                src="http://localhost/Joomla_5.0.0-alpha2-Alpha-Full_Package/images/banners/shop-ad-books.jpg#joomlaImage://local-images/banners/shop-ad-books.jpg?width=468&height=60"
                                alt="Banner"
                                width="50"                                height="50"                            >
```

After PR:
```
<img src="http://localhost/Joomla_5.0.0-alpha2-Alpha-Full_Package/images/banners/shop-ad-books.jpg" alt="Banner" width="468" height="60" loading="lazy">
```

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by wojtekLs** [comment]
@HLeithner reasons?

**Comment by HLeithner** [comment]
all pull requests have been rebased because the new branch for Joomla 5.1 has been created. since Joomla 5.0 don't get any new features from this point we we automatically rebase Pull requests with the label feature to the next minor branch. and bugs to the olders supported branch (in this case 4.4)

**Comment by hans2103** [comment]
> The width/height settings are ignored.
> 
> Before PR:
> 
> ```
>                                  src="http://localhost/Joomla_5.0.0-alpha2-Alpha-Full_Package/images/banners/shop-ad-books.jpg#joomlaImage://local-images/banners/shop-ad-books.jpg?width=468&height=60"
>                                 alt="Banner"
>                                 width="50"                                height="50"                            >
> ```
> 
> After PR:
> 
> ```
> Banner
> ```

are they? The attributes `width` and `height` are still there.

**Comment by hans2103** [comment]
please close this PR as it is replaced by https://github.com/joomla/joomla-cms/pull/42508

**Pull request closed without merging**